### ISSUE TITLE
KeyError: 'RLHF-V-Dataset'

### ISSUE THREAD

**Issue by DuoLong** [RLHF-V/RLHF-V#8]
When I run run_RLHFV.sh,
'bash ./script/train/run_RLHFV.sh ./RLHFV_checkpoints/dpo_exp master RLHFV 5.0 ./RLHF-V_SFT_weight RLHF-V-Dataset 1 320 40 0.5 False True'
the error comes:''

File "/opt/tiger/xxx/Muffin/muffin/train/train_muffin.py", line 278, in make_dpo_data_module
    train_dataset = DPODataset(tokenizer=tokenizer,
  File "/opt/tiger/xxx/Muffin/muffin/train/train_muffin.py", line 178, in __init__
    self.list_data_dict = create_multi_data_source_dataset(multimodal_cfg['data_source_names'], multimodal_cfg['data_source_weights'])
  File "/opt/tiger/xxx/Muffin/muffin/train/train_muffin.py", line 144, in create_multi_data_source_dataset
    ds = SingleDataSourceDataset(name, *register_data_path[name]())
  File "/opt/tiger/xxx/Muffin/muffin/data/data_processors.py", line 47, in __getitem__
    return self._dict[key]
KeyError: 'RLHF-V-Dataset'

Why it comes?

Thank you very much

**Comment by Mr-Loevan** [comment]
the same.

**Comment by Mr-Loevan** [comment]
@Haoye17 Could you please help solve this issue?

**Comment by Haoye17** [comment]
Hi @DuoLong @Mr-Loevan ! Very sorry for the inconvenience! We are currently reformulating our codes to make the data loading simpler. For now, please try the following modifications to see if they can help:

In `Muffin/muffin/data/data_processors.py`,
change the code from
```python
@register_data_processor('RLHF-V-Hall_v0')
def dpo_cvpr_ncrp_vqa_processor(*args, **kwargs):
    return dpo_preference_processor(*args, **kwargs)


@register_data_path('RLHF-V-Hall_v0')
def dpo_cvpr_ncrp_vqa_path():
    data_dir = pathlib.Path(__file__).parent.resolve() / '../../data/RLHF-V-Hall_v0'
    return gather_data_files_by_glob(data_dir, pattern='*dpo_with_rlhf-v-sft_logp_train-1401.tsv')
```

to

```python
@register_data_processor('RLHF-V-Dataset')
def dpo_cvpr_ncrp_vqa_processor(*args, **kwargs):
    return dpo_preference_processor(*args, **kwargs)


@register_data_path('RLHF-V-Dataset')
def dpo_cvpr_ncrp_vqa_path():
    data_dir = pathlib.Path(__file__).parent.resolve() / '../../data/RLHF-V-Dataset'
    return gather_data_files_by_glob(data_dir, pattern='*withlogp-1401.tsv')
```

If there still be some problems, feel free to ask~

**Comment by Mr-Loevan** [comment]
![image](https://github.com/RLHF-V/RLHF-V/assets/70183000/fa87a4d2-99d7-4254-b965-82229bba42b8)
@Haoye17 Hi~ Thank you for your response. I've changed the code, and launched the dpo train.
what's the possible reason? This is trained on 1 a100 80G, perhaps OutOfMemory of GPU?

**Comment by Haoye17** [comment]
Hi @DuoLong!  In our code, we use pytorch DDP for distributed training on 8 gpus by default and it is defined by `GPUS_PER_NODE=8` in the script `Muffin/script/train/ddpo_muffin_PATH.sh`. I think this error might be caused by the single gpu you are using now 🤔. Could you please try to use 8 gpus for training?

**Comment by Mr-Loevan** [comment]
Hi @Haoye17 , I've encountered this bug while gpu==2 :

![image](https://github.com/RLHF-V/RLHF-V/assets/70183000/8b8aac9d-bc22-482a-bd82-2603841c3a21)


And can LORA be applied to this codebase? 2*A100 is not enough.

**Comment by yiranyyu** [comment]
> Hi @Haoye17 , I've encountered this bug while gpu==2 :
> 
> ![image](https://private-user-images.githubusercontent.com/70183000/298440108-8b8aac9d-bc22-482a-bd82-2603841c3a21.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5MjczNDksIm5iZiI6MTcwNTkyNzA0OSwicGF0aCI6Ii83MDE4MzAwMC8yOTg0NDAxMDgtOGI4YWFjOWQtYmMyMi00ODJhLWJkODItMjYwMzg0MWMzYTIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIyVDEyMzcyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBhZGUyOTIwNjE5OWZlNjdkMGRhYjJlNThkZThmYTY0ODVkOWRkYWIzZWI0OGRhYmE5YjljMzI0ZTFjNGE1MjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.zLrelI5U-JHiNgK-VH_bfGbCu66BOj6Mht2gZHfmUPA)
> 
> And can LORA be applied to this codebase? 2*A100 is not enough.

We also encountered such warnings during development, but it seems not harmful to the result model.

Our model class inherits the `LlamaModel` class and inherently support LoRA training. We refer you to the official [training guidance](https://huggingface.co/docs/peft/conceptual_guides/lora) about LoRA for more details.

**Comment by Mr-Loevan** [comment]
> > Hi @Haoye17 , I've encountered this bug while gpu==2 :
> > ![image](https://private-user-images.githubusercontent.com/70183000/298440108-8b8aac9d-bc22-482a-bd82-2603841c3a21.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5MjczNDksIm5iZiI6MTcwNTkyNzA0OSwicGF0aCI6Ii83MDE4MzAwMC8yOTg0NDAxMDgtOGI4YWFjOWQtYmMyMi00ODJhLWJkODItMjYwMzg0MWMzYTIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIyVDEyMzcyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBhZGUyOTIwNjE5OWZlNjdkMGRhYjJlNThkZThmYTY0ODVkOWRkYWIzZWI0OGRhYmE5YjljMzI0ZTFjNGE1MjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.zLrelI5U-JHiNgK-VH_bfGbCu66BOj6Mht2gZHfmUPA)
> > And can LORA be applied to this codebase? 2*A100 is not enough.
> 
> We also encountered such warnings during development, but it seems not harmful to the result model.
> 
> Our model class inherits the `LlamaModel` class and inherently support LoRA training. We refer you to the official [training guidance](https://huggingface.co/docs/peft/conceptual_guides/lora) about LoRA for more details.

![image](https://github.com/RLHF-V/RLHF-V/assets/70183000/83e86874-56c4-4fdc-823a-6ccc76b0d83d)

I tried to use LoRA, but Multiway Network in vision tower(Beit3) don't support LoRA.

**Comment by Cooperx521** [comment]
@Mr-Loevan Hello~ have you solved this problem? I have used 8 A100, but I also encountered this problem. @yiranyyu Could you help to have a look? I have set `GPUS_PER_NODE=8`, and command line is `srun --partition=llm3 --gres=gpu:8 --nodes=1 --cpus-per-task=16 --job-name=RLHF-V --ntasks-per-node=8 bash run_RLHFV.sh`
![image](https://github.com/RLHF-V/RLHF-V/assets/131770126/0977bbda-40da-48e2-99fb-7b5e9997c628)

**Comment by yiranyyu** [comment]
> @Mr-Loevan Hello~ have you solved this problem? I have used 8 A100, but I also encountered this problem. @yiranyyu Could you help to have a look? I have set `GPUS_PER_NODE=8`, and command line is `srun --partition=llm3 --gres=gpu:8 --nodes=1 --cpus-per-task=16 --job-name=RLHF-V --ntasks-per-node=8 bash run_RLHFV.sh` ![image](https://private-user-images.githubusercontent.com/131770126/300373050-0977bbda-40da-48e2-99fb-7b5e9997c628.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDcxODg4MDMsIm5iZiI6MTcwNzE4ODUwMywicGF0aCI6Ii8xMzE3NzAxMjYvMzAwMzczMDUwLTA5NzdiYmRhLTQwZGEtNDhlMi05OWZiLTdiNWU5OTk3YzYyOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIwNlQwMzAxNDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05OGE4NjBiYWRhNzg2ZDRhZGIyNDFlMzNiZmU3OGM1YjIyYTU5ZmQ0YjBmODA0ZTU3ZmU0MWU1ZjIwYzZlYWRkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.FC_yJhzu1xuUKvcMEzsCHdGrw1YUeEuRWX1FD9gBEeM)

Without more information, we cannot give more detail suggestions about solving this problem. It seems more related to your environment.
